internal/log/logger.go

```diff
@@ -0,0 +1,62 @@
+package log
```

Comment:
Let's rename this to `internal_log` (or similar):
- It can then be in `internal_log/` directory (same level as `dat`, `kvs`, etc.)
- It conveys clearly that this is not intended to be external API, and that we're only exporting symbols for re-use internally in other packages
- Avoids clashing with `log` package of stdlib, which should be avoided (https://blog.golang.org/package-names)

Reply:
Sure, this is definitely a style thing. I come from this school https://medium.com/@benbjohnson/standard-package-layout-7cdbc8391fc1 where you package dependencies to specifically cause a collision and encourage not directly accessing them. For reference, using log and renaming / shadowing stdlog is actually the common practice among the most popular log interfaces. Examples: https://github.com/go-log/log/blob/master/log/log.go and https://github.com/go-kit/kit/tree/master/log which both cover stdlib log to discourage outside use of it.

Also, we don't want this at the same level as dat/kvs/etc intentionally. The internal/ package is being used here to to create a compile time error if any consumer of dat attempts to import this log package. Reference from official integration in 1.4 https://golang.org/doc/go1.4#internalpackages & [the proposal](https://docs.google.com/document/d/1e8kOo3r51b2BWtTs_1uADIA5djfXhPT36s6eHVRIvaU/edit)

Reply:
That's great, thanks for all the explanations! Let's leave as is then.